### ISSUE TITLE
Missing conversion/attachment of trackerHits to track

### ISSUE THREAD

**Issue by Zehvogel** [key4hep/k4EDM4hep2LcioConv#39]
It looks like the information which tracker hits belong to a track goes missing during the conversion.

E.g. when I have a file with single muons from CLD standard reconstruction in lcio REC format I can see the hits with `dumpevent`:
```
collection name : SiTracksCT
 parameters: 

--------------- print out of Track collection --------------- 

  flag:  0x80000000
     LCIO::TRBIT_HITS : +1

 [   id   ] |   type   |    d0    |  phi     | omega    |    z0     | tan lambda|   reference point(x,y,z)        |    dEdx  |  dEdxErr |   chi2   |  ndf   
------------|----------|----------|----------|----------|-----------|-----------|---------------------------------|----------|----------|-------- 
 [00000204] | 00000000 |+1.17e-02 |+9.56e-01 |-6.10e-05 |-1.032e-02 |+1.765e-01 |(+0.00e+00, +0.00e+00, +0.00e+00)|+0.00e+00 |+0.00e+00 |+1.93e+01 |   15
 errors:   +3.772722e-05 | -9.918106e-07, +3.450092e-08 | -1.920929e-11, +9.643513e-13, +2.421880e-14 | -5.997298e-08, +4.746751e-09, +2.569830e-11, +3.967462e-05 | +4.320375e-09, -1.778298e-10, -7.268266e-13, -1.044233e-06, +3.613802e-08 | 
 tracks(id): 
 hits ->[0000+223] [0000+224] [0000+225] [0000+226] [0000+179] [0000+180] [0000+181] [0000+190] [0000+191] [0000+192] 
 radius of innermost hit +3.540330e+01 / mm ,  subdetector Hit numbers : +10, +0, +0, +0, +0, +0, +0, +0, +0, +0, +0, +0
------------|----------|----------|----------|----------|-----------|-----------|---------------------------------|----------|----------|-------- 
```
when I convert the file with `lcio2edm4hep` (or when I save the reconstruction output in edm4hep format) I get the following with `podio-dump` showing me no hits
```
SiTracksCT
          id:        type:        chi2:         ndf:        dEdx:   dEdxError:radiusOfInnermostHit:
d094e81e+0           +0 +1.933914e+01          +15 +0.000000e+00 +0.000000e+00 +3.540330e+01 
      trackerHits : 
      tracks : 
      subdetectorHitNumbers : +10 +0 +0 +0 +0 +0 +0 +0 +0 +0 +0 +0 
      trackStates : +1 +1.173224e-02 +9.557508e-01 -6.101672e-05 -1.032008e-02 +1.765175e-01 -1.000000e+00 +0.000000e+00 +0.000000e+00 +0.000000e+00  +3.772722e-05|-9.918106e-07|+3.450092e-08|-1.920929e-11|+9.643513e-13|+2.421880e-14|-5.997298e-08|+4.746751e-09|+2.569830e-11|+3.967462e-05|+4.320375e-09|-1.778298e-10|-7.268266e-13|-1.044233e-06|+3.613802e-08|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|  +2 -6.976685e-13 +9.579772e-01 -6.123351e-05 -1.304482e-13 +1.763412e-01 -1.000000e+00 +2.038578e+01 +2.894500e+01 +6.241789e+00  +8.928548e-06|-3.899025e-06|+8.897434e+01|+1.470613e-10|-3.343057e-03|+9.999651e-05|+1.408484e-07|+7.741352e-06|-2.903438e-10|+9.002223e-06|+5.598554e-10|-2.419017e-03|+4.488206e-08|-3.893906e-06|+8.886209e+01|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|  +3 -6.192280e-01 +1.086633e+00 -6.101842e-05 +4.857747e-01 +1.765156e-01 -1.000000e+00 +1.119068e+03 +1.828073e+03 +3.781249e+02  +4.191457e-01|+2.885465e-04|+2.353976e-07|-5.358560e-08|-5.085683e-11|+2.423507e-14|+1.001712e-03|+2.777121e-06|-2.546074e-09|+3.152730e-01|+2.515961e-06|+2.850458e-09|-1.891274e-12|+1.848317e-04|+1.347193e-07|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|  +4 -0.000000e+00 +1.087078e+00 -6.101842e-05 +1.917562e-13 +1.765156e-01 -1.000000e+00 +1.123008e+03 +1.834237e+03 +3.798975e+02  +4.233678e-01|+2.906353e-04|+2.361239e-07|-5.395697e-08|-5.103012e-11|+2.423507e-14|+9.756010e-04|+2.769262e-06|-2.550082e-09|+3.179742e-01|+2.536789e-06|+2.864077e-09|-1.891534e-12|+1.858140e-04|+1.347358e-07|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|+0.000000e+00|  
      dxQuantities : +0 +0.000000e+00 +0.000000e+00 
```
When I get the `edm4hep::Track` from the track collection and call `.getTrackerHits()` the output is also empty.

**Comment by tmadlener** [comment]
Assuming that the hits in an `SiTrackCT` are tracks from an Si Tracker they are probably `TrackerHitPlane`. Which in turn means they will not be attached for two reasons

1. The corresponding linking is explicitly commented out: https://github.com/key4hep/k4EDM4hep2LcioConv/blob/fba56fff3e3fa9c4650eb953c3108950fb1ec55d/k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.ipp#L817-L820
2. The `edm4hep::Track` only has relations to `edm4hep::TrackerHit` but not to the others. This worked in LCIO because there the different tracker hit types formed a hierarchy with the `EVENT::TrackerHit` as a base class. This is not possible at the moment with EDM4hep.

**Comment by tmadlener** [comment]
This should be fixed with #49. Closing this for now. Feel free to re-open if things are still not working as expected.
